GliaWidgets/SecureConversations/SecureConversations.ChatWithTranscriptModel.swift

```diff
@@ -910,6 +913,58 @@ extension SecureConversations.TranscriptModel {
             }
         }
     }
+
+    /// Wraps message to determine how to replace it when it is delivered.
+    enum MessageSource: Identifiying, Equatable {
+        case api(CoreSdkClient.Message, outgoingMessage: OutgoingMessage)
+        case socket(CoreSdkClient.Message)
+
+        var id: String {
+            message.id
+        }
+
+        var message: CoreSdkClient.Message {
+            switch self {
+            case let .api(message, _):
+                return message
+            case let .socket(message):
+                return message
+            }
+        }
+
+        var outgoingMessage: OutgoingMessage? {
+            switch self {
+            case let .api(_, outgoingMessage):
+                return outgoingMessage
+            case  .socket:
+                return nil
+            }
+        }
+    }
+
+    func receiveMessage(_ message: MessageSource) {
```

Comment:
Maybe `receiveMessage(fromSource source: MessageSource)` as the previous name indicates that we are receiving a message.

Reply:
Renamed 3118054ac0dd26ec64a9ea495d2a0b338a2fb2e7.